src/groupBy.test.ts

```diff
@@ -176,6 +176,56 @@ describe("typing", () => {
     >();
   });
 
+  test("symbol", () => {
+    const mySymbol = Symbol("mySymbol");
+    const data = groupBy(
+      [
+        { [mySymbol]: "cat", b: 123 },
+        { [mySymbol]: "dog", b: 456 },
```

Comment:
I'm confused; why does it matter that the object has a prop with `mySymbol`? isn't the grouping function's return value the only thing we care about here?